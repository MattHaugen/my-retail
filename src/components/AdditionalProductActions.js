import React from 'react';
import styled from 'styled-components';
import colors from '../constants/colors';

const Container = styled.div`
  width: 100%;
  display: flex;
  margin-bottom: 30px;
`;

const Button = styled.button`
  flex-basis: 0;
  flex-grow: 1;
  width: 100%;
  padding-top: 10px;
  padding-bottom: 10px;
  border-radius: 3px;
  border: none;
  color: ${colors.nearBlack};
  background-color: ${colors.nearWhite};
  text-transform: uppercase;
  font-family: 'Arial', sans-serif;
  font-size: 12px;
  font-weight: 100;
  cursor: pointer;
  margin-right: 5px;

  &:last-of-type {
    margin-right: 0px;
  }
`;

const AdditionalProductActions = () => {

  //Leaving this here so it could potentially be powered by props in the future
  const actions = [
    {
      label: 'Add to Registry',
      action: 'A link to a function or redux action, depending on what needs to be done'
    },
    {
      label: 'Add to List',
      action: 'A link to a function or redux action, depending on what needs to be done'
    },
    {
      label: 'Share',
      action: 'A link to a function or redux action, depending on what needs to be done'
    }
  ];

  const actionComponents = actions.map((action, index) => {
    return <Button key={index}>{action.label}</Button>;
  })

  return (
    <Container>
      {actionComponents}
    </Container>
  )
}

export default AdditionalProductActions;
