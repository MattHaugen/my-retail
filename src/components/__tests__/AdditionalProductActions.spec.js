import React from 'react';
import AdditionalProductActions from '../AdditionalProductActions';
import renderer from 'react-test-renderer';

describe('AdditionalProductActions component', () => {
  it('renders correctly (snapshot)', () => {
    //ACT
    const component = renderer.create(<AdditionalProductActions />);
    const json = component.toJSON();

    //ASSERT
    expect(json).toMatchSnapshot();
  });
});
