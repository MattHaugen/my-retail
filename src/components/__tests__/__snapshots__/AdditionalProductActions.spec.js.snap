// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdditionalProductActions component renders correctly (snapshot) 1`] = `
<div
  className="sc-bdVaJa keWvrX"
>
  <button
    className="sc-bwzfXH zaifQ"
  >
    Add to Registry
  </button>
  <button
    className="sc-bwzfXH zaifQ"
  >
    Add to List
  </button>
  <button
    className="sc-bwzfXH zaifQ"
  >
    Share
  </button>
</div>
`;
